SCClassLibrary/Common/GUI/PlusGUI/Control/ServerPlusGUI.sc

```diff
@@ -1,3 +1,23 @@
+ServerTreeView {
+	*new { |aServer, bounds=(Rect(128, 64, 400, 400)), interval=0.5|
+		var onClose, window;
+		window = Window.new(name.asString, bounds, scroll:true).front;
+		onClose = aServer.plotTreeView(interval, window.view, { defer {window.close}; });
```

Comment:
could you move the code from `plotTreeView` here? This would simplify the code there, you can just check if a window exists already and if not, make it an instance of `ServerTreeView`.

Reply:
Also, the obscure `onClose` logic could finally go away.
To be backward compatible, `plotTreeView` could still return a function that closes the instance (although I'd prefer if it returned the instance… hm.)

Reply:
@telephon 
I think I can do it, could you confirm that I have understood your request correctly? Which of the two things should I do?

1. Move the `plotTreeView` method in the `Server` class to the `ServerTreeView` class. For backward compatibility, create a simplified `plotTreeView` method in the `Server` class.

2. Implement the `plotTreeView' method in the `Server' class as a function in the `.new' method of the `ServerTreeView' class. Leave the `plotTreeView` method in `Server` untouched for backward compatibility.

Reply:
I meant (1), but make it backward compatible, by making the `server.plotTreeView` return a function that works like the old one ("on close").